### PR TITLE
Added ChangeDirectoryAsync to SftpClient

### PR DESCRIPTION
**Pull request by snargledorf**

### CODE REVIEW

**Comment by snargledorf** [comment]
May be relevant for #1415

File: test/Renci.SshNet.IntegrationTests/OldIntegrationTests/SftpClientTest.ListDirectory.cs [hunk 1]
```diff
@@ -229,6 +229,75 @@ public void Test_Sftp_Change_Directory()
...
+                await sftp.ChangeDirectoryAsync("/home/sshnet/test1/test1_1/../test1_2", CancellationToken.None).ConfigureAwait(false);
+
+                Assert.AreEqual(sftp.WorkingDirectory, "/home/sshnet/test1/test1_2");
+
+                await sftp.ChangeDirectoryAsync("../../", CancellationToken.None).ConfigureAwait(false);
+
+                sftp.DeleteDirectory("test1/test1_1");
+                sftp.DeleteDirectory("test1/test1_2");
+                sftp.DeleteDirectory("test1/test1_3");
+                sftp.DeleteDirectory("test1");
```
Comment (Rob-Hague) [inline test/Renci.SshNet.IntegrationTests/OldIntegrationTests/SftpClientTest.ListDirectory.cs:293]: I assume these will become async with your other PR

File: test/Renci.SshNet.IntegrationTests/SftpTests.cs [hunk 2]
```diff
@@ -3993,6 +3992,36 @@ public void Sftp_ChangeDirectory_DirectoryDoesNotExist()
             }
         }
 
+        [TestMethod]
+        public async Task Sftp_ChangeDirectory_DirectoryDoesNotExistAsync()
```
Comment (Rob-Hague) [inline test/Renci.SshNet.IntegrationTests/SftpTests.cs:3996]: Seems like we have some duplicate/redundant tests between this and the "old" tests files 🤷

**Review by Rob-Hague — APPROVED** [review]
Thanks

**Pull request merged**